### ISSUE TITLE
TypeError when running `upload_analytic` method

### ISSUE THREAD

**Issue by mattphotonman** [voxel51/api-py#40]
```python
from voxel51.api import API
api = API()
api.upload_analytic("/path/to/analytic.json")

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-1f6d68783277> in <module>
----> 1 api.upload_analytic("/path/to/analytic.json")

~/api-py/voxel51/api.py in upload_analytic(self, doc_json_path)
    168             res = self._requests.post(
    169                 endpoint, headers=self._header, files=files)
--> 170         _validate_response(res)
    171         return _parse_json_response(res)["analytic"]
    172 

~/api-py/voxel51/api.py in _validate_response(res)
    793 def _validate_response(res):
    794     if not res.ok:
--> 795         raise APIError.from_response(res)
    796 
    797 

~/api-py/voxel51/api.py in from_response(cls, res)
    780 
    781         try:
--> 782             message = _parse_json_response(res)["error"]["message"]
    783         except ValueError:
    784             res.raise_for_status()

~/api-py/voxel51/api.py in _parse_json_response(res)
    797 
    798 def _parse_json_response(res):
--> 799     return json.loads(res.content)

/usr/lib/python3.5/json/__init__.py in loads(s, encoding, cls, object_hook, pars
e_float, parse_int, parse_constant, object_pairs_hook, **kw)
    310     if not isinstance(s, str):
    311         raise TypeError('the JSON object must be str, not {!r}'.format(
--> 312                             s.__class__.__name__))
    313     if s.startswith(u'\ufeff'):
    314         raise JSONDecodeError("Unexpected UTF-8 BOM (decode using utf-8-
sig)",

TypeError: the JSON object must be str, not 'bytes'
```

**Comment by brimoor** [comment]
Get comfortable with `str` vs `bytes` errors in Python 3. It makes a distinction between the two.

Changing `json.loads(res.content)` to `json.loads(str(res.content))` should do the trick.

**Comment by brimoor** [comment]
@mattphotonman please verify and make a PR if this fixes it for you.

**Comment by mattphotonman** [comment]
> Get comfortable with `str` vs `bytes` errors in Python 3. It makes a distinction between the two.
> 
> Changing `json.loads(res.content)` to `json.loads(str(res.content))` should do the trick.

Do we have a standard for testing code in both python2 and python3?

**Comment by mattphotonman** [comment]
Made the change and got past this error and onto the next one:
```python
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
~/api-py/voxel51/api.py in from_response(cls, res)
    781         try:
--> 782             message = _parse_json_response(res)["error"]["message"]
    783         except ValueError:

~/api-py/voxel51/api.py in _parse_json_response(res)
    798 def _parse_json_response(res):
--> 799     return json.loads(str(res.content))

/usr/lib/python3.5/json/__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    318             parse_constant is None and object_pairs_hook is None and not kw):
--> 319         return _default_decoder.decode(s)
    320     if cls is None:

/usr/lib/python3.5/json/decoder.py in decode(self, s, _w)
    338         """
--> 339         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    340         end = _w(s, end).end()

/usr/lib/python3.5/json/decoder.py in raw_decode(self, s, idx)
    356         except StopIteration as err:
--> 357             raise JSONDecodeError("Expecting value", s, err.value) from None
    358         return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

HTTPError                                 Traceback (most recent call last)
<ipython-input-3-1f6d68783277> in <module>
----> 1 api.upload_analytic("/home/matthew/theta/docker/voxel51-analytics/analytics/docs/infrastructure-sense-v0.1.json")

~/api-py/voxel51/api.py in upload_analytic(self, doc_json_path)
    168             res = self._requests.post(
    169                 endpoint, headers=self._header, files=files)
--> 170         _validate_response(res)
    171         return _parse_json_response(res)["analytic"]
    172 

~/api-py/voxel51/api.py in _validate_response(res)
    793 def _validate_response(res):
    794     if not res.ok:
--> 795         raise APIError.from_response(res)
    796 
    797 

~/api-py/voxel51/api.py in from_response(cls, res)
    782             message = _parse_json_response(res)["error"]["message"]
    783         except ValueError:
--> 784             res.raise_for_status()
    785 
    786         return cls(message, res.status_code)

~/myenv/lib/python3.5/site-packages/requests/models.py in raise_for_status(self)
    938 
    939         if http_error_msg:
--> 940             raise HTTPError(http_error_msg, response=self)
    941 
    942     def close(self):

HTTPError: 404 Client Error: Not Found for url: https://api.voxel51.com/v1analytics
```

**Comment by brimoor** [comment]
Apologies @mattphotonman, this is fixed here: https://github.com/voxel51/api-py/pull/41.

**Comment by mattphotonman** [comment]
> Apologies @mattphotonman, this is fixed here: #41.

I think that fixes one problem, but I think we also have to do `json.loads(res.content.decode("utf-8"))` instead of `json.loads(str(res.content))`.

**Comment by mattphotonman** [comment]
```python
x = b"hello"
type(x)   # 'bytes'
str(x)   # "b'hello'"
x.decode("utf-8")  # "hello"
```
